Scheduler/main.c

```diff
@@ -250,17 +261,84 @@ void printProcessStats(process* processArray, int count)
    int i;
    for (i = 0; i < count; i++)
    {
-      fprintf(outputFile, "%s wait %d turnaround %d\n", processArray[i].name,
-                                                        processArray[i].wait,
-                                                        processArray[i].endTime - processArray[i].startTime);
+      if(processArray[i].endTime > 0)
+         fprintf(outputFile, "%s wait %d turnaround %d\n", processArray[i].name,
+                                                           processArray[i].wait,
+                                                           processArray[i].endTime - processArray[i].startTime);
+      else
+         fprintf(outputFile, "%s wait %d start %d didn't finish\n", processArray[i].name,
+                                                                    processArray[i].wait,
+                                                                    processArray[i].startTime);
    }
 
 }
 
 /** Scheduling algorithms **/
 void runFCFS()
 {
+   int idxOfCurrent = -1;
+
+   // Iterate through each time slot of the total runtime.
+   int time;
+   for (time = 0; time < runtime; time++)
+   {
+      // Determine if current process has finished.
+      if ((-1 != idxOfCurrent) && (0 == processes[idxOfCurrent].burst))
```

Comment:
Because this is the only place in which you check for a process being finished, this algorithm will fail to set a process as "finished" if it happens to finish in the last time step (try running your code using set1_processes.in as input). You should add this same check right after the while loop ends to account for that situation.